mathesar_ui/src/components/EditableTextWithActions.svelte

```diff
@@ -52,7 +56,13 @@
     <TextInput on:focus={makeEditable} value={initialValue} />
   {:else}
     <div class="input-container">
-      <TextInput disabled={isSubmitting} autofocus bind:value />
+      <svelte:component
+        this={inputComponentAndProps.component}
+        {...inputComponentAndProps.props || {}}
+        autofocus
+        bind:value
+        disabled={isSubmitting}
+      />
```

Comment:
This displays a text input by default for all input types. Appropriate input should be used here instead of text inputs for all data types.

Reply:
c8d513d0